pkg/harvester/models/devices.harvesterhci.io.migconfiguration.js

```diff
@@ -0,0 +1,123 @@
+import SteveModel from '@shell/plugins/steve/steve-class';
+import { escapeHtml } from '@shell/utils/string';
+import { colorForState } from '@shell/plugins/dashboard-store/resource-class';
+
+/**
+ * Class representing vGPU MIGConfiguration resource.
+ * @extends SteveModal
+ */
+export default class MIGCONFIGURATION extends SteveModel {
+  get _availableActions() {
+    let out = super._availableActions;
+
+    out = out.map((action) => {
+      if (action.action === 'showConfiguration') {
+        return { ...action, enabled: !this.spec.enabled };
+      } else if (action.action === 'goToEditYaml') {
+        return { ...action, enabled: !this.spec.enabled };
+      } else if (action.action === 'goToEdit') {
+        // need to wait for status to be disabled or empty value, then allow user to editConfig
+        return { ...action, enabled: !this.spec.enabled && ['disabled', ''].includes(this.configStatus) };
+      } else {
+        return action;
+      }
+    });
+
+    out.push(
+      {
+        action:  'enableConfig',
+        enabled: !this.isEnabled,
+        icon:    'icon icon-fw icon-dot',
+        label:   'Enable',
+      },
+      {
+        action:  'disableConfig',
+        enabled: this.isEnabled,
+        icon:    'icon icon-fw icon-dot-open',
+        label:   'Disable',
+      },
+    );
+
+    return out;
+  }
+
+  get canYaml() {
+    return false;
+  }
+
+  get disableResourceDetailDrawer() {
+    return true;
+  }
+
+  get canDelete() {
+    return false;
+  }
+
+  get configStatus() {
+    return this.status.status;
+  }
+
+  get actualState() {
+    return this.isEnabled ? 'Enabled' : 'Disabled';
+  }
+
+  get stateDisplay() {
+    return this.actualState;
+  }
+
+  get stateColor() {
+    const state = this.actualState;
+
+    return colorForState(state);
+  }
+
+  get isEnabled() {
+    return this.spec.enabled;
+  }
+
+  get configuredProfiles() {
+    if (!this.spec?.profileSpec || this.spec.profileSpec.filter((p) => p.requested > 0).length === 0) {
+      return '';
+    }
+    const configuredProfiles = this.spec?.profileSpec.filter((p) => p.requested > 0);
+
+    return configuredProfiles.reduce((acc, profile) => {
+      const display = `${ profile.name } * ${ profile.requested }`;
+
+      if (acc !== '') {
+        acc = `${ acc }, ${ display }`;
+      } else {
+        acc = display;
+      }
+
+      return acc;
+    }, '');
+  }
```

Comment:
[Nit] This can be simplified by using `map` directly.
```js
 get configuredProfiles() {
    const configuredProfiles = this.spec?.profileSpec?.filter((p) => p.requested > 0) || [];

    if (configuredProfiles.length === 0) {
      return '';
    }

    return configuredProfiles
      .map((profile) => `${ profile.name } * ${ profile.requested }`)
      .join(', ');
  }
```